docs/adrs/0080-api-versioning-based-on-test-run.md

```diff
@@ -0,0 +1,49 @@
+
+# How to implement API Versioning for Prime API
+
+Everything in the [previous ADR](./0078-api-versioning.md) outside of the file organization strategy are accurate and reflect what is happening in our system.
+
+## File/Code Organization
+
+### Swagger Organization
+
+We are continuing to leverage [Option 2: Have an entirely separate version 2 swagger file](https://transcom.github.io/mymove-docs/docs/adrs/api-versioning#swagger-organization) from the original ADR.
+
+Going through this approach we were able to confirm the exist pros and cons.
+
+> Pros: It would be easier to keep the definitions separate and avoid accidental introduction of breaking changes. It would enable us to use the generated code for our handlers and payload/model files.
+>
+> Cons: More generated files to manage
+
+**Additional Recommendations:** Try and reference existing definitions when possible and create new versions of the definitions when something changes. This should help identify the definitions that have been modified from version to version. You can see an example in this [PR](https://github.com/transcom/mymove/pull/10952/files#diff-ccf222ad69199a6edb2088799e4ac7f8118591dc804aff747293bf1e81262940). Having to copy over existing definitions from the previous version can be tedious and if this is done in the main yaml file it can make it hard to quickly identify what fields are being modified.
+
+### Handler Organization
+
+There are no updates to this section.
+
+### Services Organization
+
+This is where the largest deviation is.
+
+Previously we were suggesting to go with [Option 3: Creating subdirectories and using flags](https://transcom.github.io/mymove-docs/docs/adrs/api-versioning#services-organization). This approach had the following pros and cons:
+
+> Pros: Clear delineation between the two versions. There would not need to be a break from how we are currently utilizing our services and interfaces. We would only need to pull in the changed services into our subdirectories, which would lead to less code duplication.
+>
+> Cons: Repetitive code. Naming questions arise, in order to delineate the new from the old. Difficulty when dealing with more complicated services, especially when we use a service in another service.
+
+Through this test run, we realized that the flagging system had an additional con in that it was obfuscating complexity and enabling folks to potentially misuse the flag. The default behavior also made it so that all existing tests that used a given service would have been updated to leverage the new versioned service.
```

Comment:
**questions:** 
What do you mean by default behavior here? 
Can you clarify what you mean by? 
> all existing tests that used a given service would have been updated to leverage the new versioned service.

Reply:
Thanks for calling this out. I'll try to make this more clear in the ADR as well.

If you take a look at the [example](https://github.com/transcom/mymove/compare/0c4be3b...8ad4aed) in the ADR for this choice and look at the `pkg/services/move_task_order/move_task_order_fetcher.go` file you will see that the `ListPrimeMoveTaskOrders` function tries to call the v1 or v2 service based on the status of the flags, but defaults to calling the v2 service if it fails both of the conditionals. For tests, the way that the flag was setup meant that we would always fail both of the conditionals and call the v2 service. 

Did that help clarify things?

Reply:
I modified this portion to be: https://github.com/transcom/mymove-docs/pull/384/commits/d68616e32f937ed126d7fcbe2895c1d795133196. Let me know if you think this clarifies things.